### PR TITLE
task 8 completed

### PR DESCRIPTION
**Pull request by Glt1b**

### CODE REVIEW

File: controllers/games.js [hunk 1]
```diff
@@ -52,3 +52,12 @@ exports.getCommentsByReviewId = (req, res, next) => {
 
 };
 
+exports.updateReview = (req, res, next) => {
+    setReview(req.params.review_id, req.body).then((result) => {
+        console.log(result)
```
Comment (hevdevs) [inline controllers/games.js:57]: remove console.log before merge

File: models/games_models.js [hunk 2]
```diff
@@ -44,5 +44,33 @@ exports.selectCommentsByReviewId = (review_id) => {
...
+            WHERE review_id = $1;`, [review_id]).then((result) => {
+                const newVotes = result.rows[0].votes + body.inc_votes
+                
+                return db.query(`
+                UPDATE reviews
+                SET votes = $1
+                WHERE review_id = $2
+                RETURNING *;`, [newVotes, review_id])
+            }).then((result) => {
+                console.log(result.rows[0])
```
Comment (hevdevs) [inline models/games_models.js:95]: remove console.log before merge

File: app.js [hunk 3]
```diff
@@ -21,6 +22,7 @@ app.use((err, req, res, next) => {
     res.status(err.status).send({ msg: err.msg });
   }
   else if (err.code === '22P02') {
+    console.log(err)
```
Comment (hevdevs) [inline app.js:25]: remove console.log before merge

File: __tests__/app.test.js [hunk 4]
```diff
@@ -119,4 +119,59 @@ describe('GET/api/reviews', () => {
...
+      const update = {inc_votes: 5}
+      return request(app)
+        .patch('/api/reviews/2')
+        .send(update)
+        .expect(201)
+        .then((res) => {
+          expect(res.body.review.votes).toEqual(10)
+          expect(res.body.review).toMatchObject({
+            created_at: expect.any(String),
+            category: expect.any(String),
```
Comment (hevdevs) [inline __tests__/app.test.js:135]: If we want to expect the object to look a certain way here in our tests - we can expect the exact values for the given review. E.g. we don't expect the category for review 2 to be ANY string, we know that review 2's category is 'dexterity'. The main thing to be checking here though is that the votes for the specified review have increased which you have done 👍

File: __tests__/app.test.js [hunk 5]
```diff
@@ -119,4 +119,59 @@ describe('GET/api/reviews', () => {
...
+        .expect(400)
+        .then((res) => {
+          expect(res.body.msg).toEqual('Bad Request');
+        });
+    });
+  
+    test('400 - Bad request when pass invalid comment object', () => {
+      const update = {}
+      return request(app)
+        .patch('/api/reviews/2')
```
Comment (hevdevs) [inline __tests__/app.test.js:170]: What if we send an object in the format {inc_votes: "not-a-number"} ? Or {wrong_key: 1} ? 
Add some 400 tests for these too not just an empty object

**Review by hevdevs — CHANGES_REQUESTED** [review]
Just a couple extra tests needed for the request body - please put in another PR when done

File: models/games_models.js [hunk 6]
```diff
@@ -58,17 +85,18 @@ exports.setReview= (review_id, body) => {
...
                 const newVotes = result.rows[0].votes + body.inc_votes
-                
+
                 return db.query(`
                 UPDATE reviews
                 SET votes = $1
                 WHERE review_id = $2
                 RETURNING *;`, [newVotes, review_id])
             }).then((result) => {
                 console.log(result.rows[0])
```
Comment (hevdevs) [inline models/games_models.js:95]: remove console.log before merge
